README.md

```diff
@@ -5,45 +5,164 @@
 
 <img src="fuse/doc/FuseMedML-logo.png" alt="drawing" width="30%"/>
 
-# What is FuseMedML?
-FuseMedML is an open-source python-based framework designed to enhance collaboration and accelerate discoveries in **F**used **M**edical data through advanced **M**achine **L**earning technologies. Initial version is PyTorch-based and focuses on deep learning on medical imaging.
+# Effective Code Reuse across ML projects!
+**Battle tested in the medical domain:** Medical Imaging, Computational Molecular Biochem and EHR Clinical Data Analysis
+
+Skip to [install instructions section](#Installation)  
+Skip to [complete code examples](#fusemedml-from-the-ground-up)
+
+# Motivation - *"*Oh, the pain!*"*
+Analyzing **many** ML research projects we discovered that 
+* Projects bring up is taking **far too long**, even when very similar projects were already done in the past by the same lab!
+* Porting individual components across projects was *painful* - resulting in **"reinventing the wheel" time after time**
+
+# How the magic happens
+
+## 1. A simple yet super effective design concept
+### Data is kept in a nested (hierarchical) dictionary
+This is a key aspect in FuseMedML (shortly named as "fuse"). It's a key driver of flexiblity, and allows to easily deal with multi modality information. 
+```python
+from fuse.utils import NDict
+
+sample_ndict = NDict()
+sample_ndict['input.mri'] = # ...
+sample_ndict['input.ct_view_a'] = # ...
+sample_ndict['input.ct_view_b'] = # ...
+sample_ndict['groundtruth.disease_level_label'] = # ...
+```
 
+This data can be a single sample, it can be for a minibatch, for an entire epoch, or anything that is desired.
+The "nested key" ("a.b.c.d.etc') is called "path key", as it can be seen as a path inside the nested dictionary.
 
-# Why use FuseMedML?
-Successful deep learning R&D must rely on knowledge and experiments, accumulated over a wide variety of projects, and developed by different people and teams.
+**Components are written in a way that allows to define input and output keys, to be read and written from the nested dict**  
+See a short introduction video (3 minutes) to how FuseMedML components work:  
 
-FuseMedML is an outstanding collaboration framework that allows you to rerun an experiment or reuse some of the capabilities originally written for different projects—all with minimal effort.\
-Using FuseMedML, you can write generic components that can be easily shared between projects in a plug & play manner, simplifying sharing and collaboration.
+https://user-images.githubusercontent.com/7043815/177197158-d3ea0736-629e-4dcb-bd5e-666993fbcfa2.mp4
 
-The framework’s unique software design provides many advantages, making it an ideal framework for deep-learning research and development in medical imaging:
 
-* **Rapid development** -
+### Examples - using FuseMedML-style components
```

Comment:
Ok, this is it, your chance to show the readers why fuse is the awesomest thing ever. But you need to be careful. Showing new syntactical requirements without showing their payoff is not convincing. 
Learning to encode things with a new `NDict` format, wrapping losses and metrics in fuse classes--these are not features, they are costs for the dev. The feature is what they enable. 

When do we get to see that? Can we see it sooner 😃 ?